contracts/RelayerHub.sol

```diff
@@ -8,88 +8,182 @@ import "./System.sol";
-      uint256 newRequiredDeposit = BytesToTypes.bytesToUint256(32, value);
-      require(newRequiredDeposit > 1 && newRequiredDeposit <= 1e21 && newRequiredDeposit > dues, "the requiredDeposit out of range");
-      requiredDeposit = newRequiredDeposit;
-    } else if (Memory.compareStrings(key,"dues")) {
-      require(value.length == 32, "length of dues mismatch");
-      uint256 newDues = BytesToTypes.bytesToUint256(32, value);
-      require(newDues > 0 && newDues < requiredDeposit, "the dues out of range");
-      dues = newDues;
-    } else {
-      require(false, "unknown param");
-    }
-    emit paramChange(key, value);
-  }
-
-  function isRelayer(address sender) external override view returns (bool) {
-    return relayersExistMap[sender];
-  }
+contract RelayerHub is IRelayerHub, System, IParamSubscriber {
+    using SafeMath for uint256;
+
+    uint256 public constant INIT_REQUIRED_DEPOSIT = 1e20;
+    uint256 public constant INIT_DUES = 1e17;
+    address public constant WHITELIST_1 = 0xb005741528b86F5952469d80A8614591E3c5B632;
+    address public constant WHITELIST_2 = 0x446AA6E0DC65690403dF3F127750da1322941F3e;
+
+    uint256 public requiredDeposit;
+    uint256 public dues;
+
+    mapping(address => admin) admins;
+    mapping(address => bool) relayAdminsExistMap;
+    mapping(address => address) adminsAndRelayers;
+    mapping(address => bool) relayerExistsMap;
+
+    struct admin {
+        uint256 deposit;
+        uint256 dues;
+    }
+
+    modifier onlyNonRegisteredAdmin() {
+        require(relayAdminsExistMap[msg.sender], "admin does not exist");
+        _;
+    }
+
+    modifier onlyRegisteredAdmin() {
+        require(relayAdminsExistMap[msg.sender], "admin does not exist");
+        require(admins[msg.sender], "admin not registered");
+        _;
+    }
+
+    modifier notContract() {
+        require(!isContract(msg.sender), "contract is not allowed to be a relayer");
+        _;
+    }
+
+    modifier noProxy() {
+        require(msg.sender == tx.origin, "no proxy is allowed");
+        _;
+    }
+
+    modifier noExist() {
+        require(!relayerExistsMap[msg.sender], "relayer already exists");
+        _;
+    }
+
+    event relayerRegister(address _relayer);
+    event relayerUnRegister(address _relayer);
+    event paramChange(string key, bytes value);
+
+    event removeAdminAddress(address _removedAdmin);
+    event addAdminAddress(address _addedAdmin);
+    event registerAdmin(address _registeredAdmin);
+    event addRelayer(address _relayerToBeAdded);
+    event removeRelayer(address _removedRelayer);
+
+
+    function init() external onlyNotInit {
+        requiredDeposit = INIT_REQUIRED_DEPOSIT;
+        dues = INIT_DUES;
+        alreadyInit = true;
+
+        // todo initialise the currently existing admins and their relayer keys
+
+        admins[WHITELIST_1] = admin(requiredDeposit, dues);
+        admins[WHITELIST_2] = admin(requiredDeposit, dues);
```

Comment:
This is inside the `init()` function which will be executed only during the deployment time to have a smooth transfer of logic since these are already doing relay work now. Or is there a flaw in this understanding?

Reply:
Oh I thought they were relayers before the hotfix was made after the hack i.e. they already deposited 100BNB before October?

Reply:
I'm pretty sure those hardcoded relayers had already deposited.  And if the balance of this smart contract address has enough tokens, it should be fine to allow them to unregister through their associated Manager key now.